File: app/src/main/java/jhulang/healthin/com/MainActivity.java
package jhulang.healthin.com;

import android.os.Bundle;
import android.support.annotation.IdRes;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Gravity;
import android.view.Menu;
import android.view.MenuItem;

import com.mikepenz.materialdrawer.Drawer;
import com.mikepenz.materialdrawer.accountswitcher.AccountHeader;
import com.mikepenz.materialdrawer.model.PrimaryDrawerItem;
import com.mikepenz.materialdrawer.model.ProfileDrawerItem;
import com.roughike.bottombar.BottomBar;
import com.roughike.bottombar.OnTabSelectListener;



import jhulang.healthin.com.fragment.ChatFragment;
import jhulang.healthin.com.fragment.HistoryFragment;
import jhulang.healthin.com.fragment.HomeFragment;
import jhulang.healthin.com.fragment.MyProfileFragment;
import jhulang.healthin.com.fragment.NewsFragment;


public class MainActivity extends AppCompatActivity {
    private Drawer.Result navigationDrawer;
    private AccountHeader.Result headerNavigation;
    private BottomBar bottomBar;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        headerNavigation = new AccountHeader()
                .withActivity(this)
                .withCompactStyle(false)
                .withSavedInstance(savedInstanceState)
                .withHeaderBackground(R.color.colorPrimary)
                .addProfiles(new ProfileDrawerItem().withName("Ridho Jhulang Aqsha").withEmail("aqsharidho@gmail.com").withIcon(getResources().getDrawable(R.drawable.ic_account2))
                )
                .build();

        navigationDrawer = new Drawer()
                .withActivity(this)
                .withToolbar(toolbar)
                .withDisplayBelowToolbar(false)
                .withActionBarDrawerToggleAnimated(true)
                .withDrawerGravity(Gravity.LEFT)
                .withSavedInstance(savedInstanceState)
                .withAccountHeader(headerNavigation)
                .withSelectedItem(0)
                .build();

        navigationDrawer.addItem(new PrimaryDrawerItem().withName("History").withIcon(getResources().getDrawable(R.drawable.ic_history)));
        navigationDrawer.addItem(new PrimaryDrawerItem().withName("About").withIcon(getResources().getDrawable(R.drawable.ic_about)));
        navigationDrawer.addItem(new PrimaryDrawerItem().withName("Exit").withIcon(getResources().getDrawable(R.drawable.ic_exit)));

        bottomBar = (BottomBar)findViewById(R.id.Bottom_bar);
        bottomBar.setOnTabSelectListener(new OnTabSelectListener() {
            Fragment fragment = null;
            @Override
            public void onTabSelected(@IdRes int tabId) {
                if (tabId == R.id.tab_home){
                    fragment = new HomeFragment();
                }else if (tabId == R.id.tab_news){
                    fragment = new NewsFragment();
                }else if(tabId == R.id.tab_chat){
                    fragment = new ChatFragment();
                }else if (tabId == R.id.tab_history){
                    fragment =  new HistoryFragment();
                }else if (tabId == R.id.tab_account){
                    fragment = new MyProfileFragment();
                }
                getSupportFragmentManager()
                        .beginTransaction()
                        .replace(R.id.content, fragment)
                        .commit();
            }
        });

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/jhulang/healthin/com/MainActivity.kt
package jhulang.healthin.com

import android.os.Bundle
import android.support.annotation.IdRes
import android.support.design.widget.NavigationView
import android.support.v4.app.Fragment
import android.support.v4.view.GravityCompat
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.CardView
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import com.roughike.bottombar.BottomBar
import com.roughike.bottombar.OnTabSelectListener
import jhulang.healthin.com.fragment.*
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {

    private var bottomBar: BottomBar? = null
    private var cardView: CardView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val toggle = ActionBarDrawerToggle(
                this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()

        nav_view.setNavigationItemSelectedListener(this)

        cardView = findViewById(R.id.CardView)
        cardView!!.setOnTouchListener


        bottomBar = findViewById(R.id.Bottom_bar)
        bottomBar!!.setOnTabSelectListener(object : OnTabSelectListener {
            internal var fragment: Fragment? = null
            override fun onTabSelected(@IdRes tabId: Int) {
                if (tabId == R.id.tab_home) {
                    fragment = HomeFragment()
                } else if (tabId == R.id.tab_news) {
                    fragment = NewsFragment()
                } else if (tabId == R.id.tab_chat) {
                    fragment = ChatFragment()
                } else if (tabId == R.id.tab_history) {
                    fragment = HistoryFragment()
                } else if (tabId == R.id.tab_account) {
                    fragment = MyProfileFragment()
                }
                supportFragmentManager
                        .beginTransaction()
                        .replace(R.id.content, fragment!!)
                        .commit()
            }
        })


    }
    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }
    override fun onNavigationItemSelected(item: MenuItem): Boolean {

        when (item.itemId) {
            R.id.nav_history -> {

            }
            R.id.nav_about -> {

            }
            R.id.nav_exit -> {

            }
            R.id.nav_manage -> {

            }
            R.id.nav_share -> {

            }
            R.id.nav_send -> {

            }
        }

        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }
}
